pandas/core/indexes/base.py

```diff
@@ -3117,35 +3466,21 @@ def symmetric_difference(self, other, result_name=None):
             attribs['freq'] = None
         return self._shallow_copy_with_infer(the_diff, **attribs)
 
-    def _get_unique_index(self, dropna=False):
-        """
-        Returns an index containing unique values.
-
-        Parameters
-        ----------
-        dropna : bool
-            If True, NaN values are dropped.
-
-        Returns
-        -------
-        uniques : index
-        """
-        if self.is_unique and not dropna:
-            return self
-
-        values = self.values
-
-        if not self.is_unique:
-            values = self.unique()
+    def _assert_can_do_setop(self, other):
+        if not is_list_like(other):
+            raise TypeError('Input must be Index or array-like')
+        return True
 
-        if dropna:
-            try:
-                if self.hasnans:
-                    values = values[~isna(values)]
-            except NotImplementedError:
-                pass
+    def _convert_can_do_setop(self, other):
+        if not isinstance(other, Index):
+            other = Index(other, name=self.name)
+            result_name = self.name
+        else:
+            result_name = get_op_result_name(self, other)
+        return other, result_name
 
-        return self._shallow_copy(values)
+    # --------------------------------------------------------------------
```

Comment:
would move these near the ssection you currently have named indexer

Reply:
done